### PR TITLE
Create data-types.md

### LINKED ISSUE #4675: [Concept Entry] MySQL data-types
### Reason/inspiration (optional)

We would like a new entry on the `Data types` concept under MySQL. The entry should go in a new file under `docs/content/mysql/concepts/data-types/data-types.md`.

The entry should include:

- An introductory paragraph that describes the concept
- A ‘Syntax’ section that includes the syntax for the concept (if applicable)
- An ‘Example’ section that shows the concept in use

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by A1cX**
data types md file

### Description
PR is linked to MySQL data-types #4675 issue assigned to me and it is an .md file describing usage, syntax and examples of various MySQL data types.

### Issue Solved
Closes #4675

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/mysql/concepts/data-types/data-types.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+);
+
+Each person will get a unique number called user_id that is increased automatically: user_id INT AUTO_INCREMENT PRIMARY KEY
+
+The fingerprint code is stored in this template, and it’s always 512 bytes long: fingerprint_template BINARY(512)
+
+Next when someone scans their fingerprint, a special code is created and stored in the database by means of converting converting the hexadecimal (base-16) string representation into its binary (base-2) equivalent:
+
+INSERT INTO biometric_data (fingerprint_template)
+VALUES (UNHEX('AABBCCDDEEFF00112233445566778899AABBCCDDEEFF00112233445566778899...'));
```
Comment (cigar-galaxy82) [inline content/mysql/concepts/data-types/data-types.md:117]: keep the above code with the above code in a code block

**Review by cigar-galaxy82 — CHANGES_REQUESTED** [review]

**Comment by cigar-galaxy82** [comment]
Hi, @A1cX thank you for contributing to Codecademy docs, the entry is well written. I have suggested a few changes please modify

File: content/mysql/concepts/data-types/data-types.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+- ‘LONGBLOB’: is used to store binary data that is beyond the capacity of other options. It can store up to 4GB and is an efficient storage option for handling complex binary objects.
+- ‘LONGTEXT’: is used to store text data that us very large, up to 4 GB of data being suitable for applications, document repositories that handle large textual content
+- ‘MEDIUMBLOB’: is used for storing large binary object data type like medium-sized images of up to 16 MB.
+- ‘MEDIUMTEXT’: is used for storing medium-sized text data of up to 16 MB and it uses more storage space than smaller text types like ’TINYTEXT’.
+- ‘SET’(val1, val2, …): is used for storing a set of values chosen from a predefined list of possible values where each ‘set’ value is stored as an integer.
+- ‘TEXT’: is used for storing medium to large text strings up to 64 KB of text data
+- ‘VARCHAR’: is used for storing variable-length strings with a maximum size specified and it gives flexibility in cases when user input to a database are needed.
```
[the remark this answers is not in the document]
Comment (A1cX) [reply content/mysql/concepts/data-types/data-types.md:66]: note added, and VARBINARY, TINYBLOB and TINYTEXT

File: content/mysql/concepts/data-types/data-types.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+Next, MySQL extracts the data and keeps record of the payments:
+
+INSERT INTO payments (user_id, order_id, payment_amount, payment_date)
+VALUES
+(10000000001, 50000000001, 999999999999, '2024-07-10 12:00:00'),
+(10000000002, 50000000002, 150000000000, '2024-07-10 12:30:00'),
+(10000000003, 50000000003, 200000000000, '2024-07-10 13:00:00');
+
+Numeric, date and time data types used in above scenario makes it possible to have recorded information stored in the database like for example:
```
[the remark this answers is not in the document]
Comment (A1cX) [reply content/mysql/concepts/data-types/data-types.md]: here I am not sure the table is formatted right, something doesn't look well

**Comment by A1cX** [comment]
file corrected

**Review by cigar-galaxy82 — APPROVED** [review]

**Comment by A1cX** [comment]
Hello, should I resolve something in this request? Thank you

**Comment by cigar-galaxy82** [comment]
@A1cX there will be a next review after which the entry will be merged

**Review by Sriparno08 — APPROVED** [review]
Looks good, @A1cX!

**Comment by A1cX** [comment]
@cigar-galaxy82  & @cigar-galaxy82  thank you guys for help with this one

**Pull request merged**